util/stopper.go

```diff
@@ -162,13 +162,20 @@ func (s *Stopper) SetStopped() {
 	}
 }
 
-// Quiesce moves the stopper to state draining, waits until all tasks
-// complete, then moves back to non-draining state. This is used from
+// Quiesce moves the stopper to state draining. Unquiesce() needs to
+// be called before the stopper is stopped. This is used from
 // unittests.
 func (s *Stopper) Quiesce() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.draining = 1
+	atomic.StoreInt32(&s.draining, 1)
```

Comment:
Now that you're correctly setting draining using an atomic op, there's nothing here that requires you get the mutex lock. Same thing goes for `Unquiesce()`.